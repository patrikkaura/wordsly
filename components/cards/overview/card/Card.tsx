import { memo, useMemo } from 'react';
import {
  Box,
  Center,
  StatGroup,
  Stat,
  StatLabel,
  StatNumber,
  StatHelpText,
} from '@chakra-ui/react';
import Avatar from 'boring-avatars';
import Link from 'next/link';

import { AVATAR_COLORS } from '@config';

import CardStatBar from './CardStatBar';

type Props = {
  id: string;
  name: string;
  rating: number;
  numberOfWords: number;
};

function Card(props: Props) {
  const { id, name, rating, numberOfWords } = props;

  const numberOfWordsLabel = useMemo(() => {
    if (numberOfWords === 1) {
      return '1 slovo';
    } else if (numberOfWords < 4) {
      return `${numberOfWords} slova`;
    }
    return `${numberOfWords} slov`;
  }, [numberOfWords]);

  return (
    <Center data-testid="word-card-container" borderWidth={1} w="270px">
      <Box>
        <Avatar
          square
          size={120}
          name={name}
          variant="beam"
          colors={AVATAR_COLORS}
        />
      </Box>
      <Box pl={2} display="flex" justifyContent="center" w="100%">
        <StatGroup>
          <Stat>
            <StatLabel>
              <Link href={`/cards/${id}`}>{name}</Link>
            </StatLabel>
            <StatNumber fontSize={20}>{numberOfWordsLabel}</StatNumber>
            <StatHelpText data-testid="word-card-stars-container">
              <CardStatBar id={id} rating={rating} />
            </StatHelpText>
          </Stat>
        </StatGroup>
      </Box>
    </Center>
  );
}

export default memo(Card);
