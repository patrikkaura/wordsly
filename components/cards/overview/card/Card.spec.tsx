import { render, screen } from '@testing-library/react';
import Card from './Card';

describe('Card', () => {
  let props = {
    id: '1',
    name: 'Card 1',
    rating: 3,
    numberOfWords: 100,
  };

  it('should match snapshot', () => {
    render(<Card {...props} />);

    expect(screen.getByTestId('word-card-container')).toMatchSnapshot();
  });

  it('render zero stars', () => {
    render(<Card {...props} rating={0} />);

    expect(screen.getByTestId('word-card-stars-container')).toMatchSnapshot();
  });

  it('should render five stars', () => {
    render(<Card {...props} rating={5} />);

    expect(screen.getByTestId('word-card-stars-container')).toMatchSnapshot();
  });
});
