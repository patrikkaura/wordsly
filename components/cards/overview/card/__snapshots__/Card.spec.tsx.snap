// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card render zero stars 1`] = `
<dd
  class="chakra-stat__help-text css-0"
  data-testid="word-card-stars-container"
>
  <svg
    class="chakra-icon css-gtaqdh"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
      fill="currentColor"
    />
  </svg>
  <svg
    class="chakra-icon css-gtaqdh"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
      fill="currentColor"
    />
  </svg>
  <svg
    class="chakra-icon css-gtaqdh"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
      fill="currentColor"
    />
  </svg>
  <svg
    class="chakra-icon css-gtaqdh"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
      fill="currentColor"
    />
  </svg>
  <svg
    class="chakra-icon css-gtaqdh"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
      fill="currentColor"
    />
  </svg>
</dd>
`;

exports[`Card should match snapshot 1`] = `
<div
  class="css-k0azss"
  data-testid="word-card-container"
>
  <svg
    fill="none"
    height="120"
    role="img"
    viewBox="0 0 36 36"
    width="120"
    xmlns="http://www.w3.org/2000/svg"
  >
    <mask
      height="36"
      id="mask__beam"
      maskUnits="userSpaceOnUse"
      width="36"
      x="0"
      y="0"
    >
      <rect
        fill="#FFFFFF"
        height="36"
        width="36"
      />
    </mask>
    <g
      mask="url(#mask__beam)"
    >
      <rect
        fill="#9ADCB9"
        height="36"
        width="36"
      />
      <rect
        fill="#FAFCD3"
        height="36"
        rx="36"
        transform="translate(-5 9) rotate(129 18 18) scale(1)"
        width="36"
        x="0"
        y="0"
      />
      <g
        transform="translate(-1 4.5) rotate(9 18 18)"
      >
        <path
          d="M13,19 a1,0.75 0 0,0 10,0"
          fill="#000000"
        />
        <rect
          fill="#000000"
          height="2"
          rx="1"
          stroke="none"
          width="1.5"
          x="10"
          y="14"
        />
        <rect
          fill="#000000"
          height="2"
          rx="1"
          stroke="none"
          width="1.5"
          x="24"
          y="14"
        />
      </g>
    </g>
  </svg>
  <div
    class="chakra-stat__group css-1gm1wx9"
    role="group"
  >
    <div
      class="chakra-stat css-1mbo1ls"
    >
      <dl>
        <dt
          class="chakra-stat__label css-0"
        >
          <a
            href="/cards/1"
          >
            Card 1
          </a>
        </dt>
        <dd
          class="chakra-stat__number css-1mr47i9"
        >
          100
           words
        </dd>
        <dd
          class="chakra-stat__help-text css-0"
          data-testid="word-card-stars-container"
        >
          <svg
            class="chakra-icon css-y7opgn"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
              fill="currentColor"
            />
          </svg>
          <svg
            class="chakra-icon css-y7opgn"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
              fill="currentColor"
            />
          </svg>
          <svg
            class="chakra-icon css-y7opgn"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
              fill="currentColor"
            />
          </svg>
          <svg
            class="chakra-icon css-gtaqdh"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
              fill="currentColor"
            />
          </svg>
          <svg
            class="chakra-icon css-gtaqdh"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
              fill="currentColor"
            />
          </svg>
        </dd>
      </dl>
    </div>
  </div>
</div>
`;

exports[`Card should render five stars 1`] = `
<dd
  class="chakra-stat__help-text css-0"
  data-testid="word-card-stars-container"
>
  <svg
    class="chakra-icon css-y7opgn"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
      fill="currentColor"
    />
  </svg>
  <svg
    class="chakra-icon css-y7opgn"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
      fill="currentColor"
    />
  </svg>
  <svg
    class="chakra-icon css-y7opgn"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
      fill="currentColor"
    />
  </svg>
  <svg
    class="chakra-icon css-y7opgn"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
      fill="currentColor"
    />
  </svg>
  <svg
    class="chakra-icon css-y7opgn"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
      fill="currentColor"
    />
  </svg>
</dd>
`;
